### DISCUSSION TITLE
crypto.getRandomValues() not supported while using aws-sdk-v3

### DISCUSSION THREAD

**Discussion by rahanid** [aws/aws-sdk-js#4474] [Q&A]
I am upgrading my nodejs typescript application from aws-sdk v2 to v3. This is a backend application which gets deployed as lambda. When ever i do a dynamo db operation to query using PK and SK , I get  "crypto.randomValues() not supported" error in logs of the lambda function.

I have tried importing 'react-native-get-random-values' package too. It didn't help.

**Comment by ajredniwja** [comment]
Hi @rahanid, 

With limited information the issue seems to be same issue as mentioned here: https://github.com/aws/aws-sdk-js-v3/discussions/3950 where user imports a pollyfill for crypto module, which I believe should resolve the issue. 

Please reach out if that doesnt work and maybe share the version you are using for node and SDK along with the code you are using. 

Thankyou

**Comment by rahanid** [comment]
@ajredniwja  I am using node 18. Have tried it with node 16 as well. Also I tried including 'react-native-get-random-values' in my package.json and tried importing the same in the first line of code where the execution begins. That still doesn't solve the problem for me.
